### PR TITLE
A audit_compare() method has been added so that you can override exactly how model updates are audited.

### PR DESCRIPTION
**Pull request by msb**

### CODE REVIEW

**Comment by rjw57** [comment]
Builds are failing :(.

File: automationcommon/models.py [hunk 1]
```diff
@@ -120,20 +120,35 @@ def _dict(self):
         # NOTE: an internal attribute has been used when introspecting the model.
         return model_to_dict(self, fields=[field.name for field in self._meta.fields])
 
+    def audit_compare(self, field, old, new):
```
Comment (rjw57) [inline automationcommon/models.py:123]: Could we not make this a method on the [Meta class](https://docs.djangoproject.com/en/2.0/topics/db/models/#meta-options)? Or, ideally, something like:
```python
class MyModel(ModelChangeMixin, models.Model)
    some_crazy_field = WeirdField()

    class Meta:
        def audit_compare_some_crazy_field(self, old, new):
            """WeirdField always stores the old value as a reversed list..."""
            return reverse(old) == new
```
where the field name is used to automatically find the right method.

If such automatic magicness is too much of a pain, I'd still prefer this to be on the Meta class.

**Review by rjw57 — CHANGES_REQUESTED** [review]
I think the feature would be nicer to use if the specialisation was on the Meta class.

File: automationcommon/models.py (hunk 1 above)
Comment (msb) [reply automationcommon/models.py:123]: I was broadly led by this advise: https://code.djangoproject.com/ticket/5793

Comment (rjw57) [reply automationcommon/models.py:123]: OK

**Review by rjw57 — APPROVED** [review]

**Pull request merged**